### PR TITLE
feat/personal-home-ui (Work in Progress)

### PR DESCRIPTION
**Pull request by jonathanprozzi**
## Considerations & Notes

- Data is mocked for building out the UI -- tried to predict the shapes based on some of our other data, but I know this'll need to be updated (including the types)
  -  For the time remaining in the vote, I figured we'd likely use an `endDate` and calculate time remaining from there (I added arbitrary dates for testing and my conversion likely doesn't consider timezone preferences, but wanted to start mocking it)
  - I was using the statuses on the existing `Proposal` type but can update depending on data structure
- Currently following the pattern:
  -  `page.tsx` -> `component.tsx` -> individual feeds such as proposals / requests -> individual components
  - Will want to move to pattern where data is fetched in the particular feed, such as the proposals request in the proposal feed
- The user profile data (avatar & username) in the header are fetched client side, but will want to move this to server to avoid the load state (will want to pass the connected user's address via cookie to the server)
- Space data (name and avatar) are placeholders -- space name is in the mock data, and the avatar isn't accurate. Will resolve this once we have the backend ready to go

I'll keep updating these notes and the Next Steps / TODOs as I go!

## Next Steps & TODOs

- [x] Add filter menu to the For You UI
- [ ] Review naming structure and potentially refactor if too verbose or inaccurate (mismatch between route and partials namespace at the moment)
- [ ] Move the data fetching for the user's profile to the server -- test out passing in connected user's address via cookie
  - [x] Look into cookies/Iron Session/SIWE for options
- [ ] Update the Sidebar to match the most recent design iteration
- [ ] Lock in the routing and match the partials folder namespace to this
- [ ] Continue to iterate with new designs and fix known issues:
  - [x] Internal scroll for the feed -- need to adjust since it's stuck at the `h-screen` and should scroll all the way to end of content

### CODE REVIEW

File: apps/web/partials/personal-home/personal-home-header.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    queryFn: async () => {
+      if (!address) return null;
+      return await subgraph.fetchProfile({ address, endpoint: config.subgraph });
+    },
+  });
+
+  return data ? data[1] : null;
+}
+
+export function PersonalHomeHeader() {
```
Comment (jonathanprozzi) [inline apps/web/partials/personal-home/personal-home-header.tsx:27]: This'll need to be refactored to not be a client side fetch and to better leverage loading states -- I pulled this from the navbar but it's a really visible load state that has a big jump to it. Will definitely improve

**Comment by jonathanprozzi** [comment]
@baiirun 

Regarding this TODO: "Move the data fetching for the user's profile to the server -- test out passing in connected user's address via cookie" 
  - I've been looking into/testing a few approaches that are cookie based, and have found a few examples of folks using Sign in with Ethereum (SIWE)+ Iron Session + app router to have the user data accessible on the server via cookies in a secure way
  - Thinking if we don't want to use SIWE we could potentially use Iron Session to encrypt/decrypt the cookie with a secret
  - Still doing a bit of testing with this since it's something we'll want to use in all user-scoped data fetching in the app router

**Comment by baiirun** [comment]
> @baiirun
> 
> Regarding this TODO: "Move the data fetching for the user's profile to the server -- test out passing in connected user's address via cookie"
> 
> * I've been looking into/testing a few approaches that are cookie based, and have found a few examples of folks using Sign in with Ethereum (SIWE)+ Iron Session + app router to have the user data accessible on the server via cookies in a secure way
> * Thinking if we don't want to use SIWE we could potentially use Iron Session to encrypt/decrypt the cookie with a secret
> * Still doing a bit of testing with this since it's something we'll want to use in all user-scoped data fetching in the app router

Since all data in Geo is public I don't know if we need to worry too much about encrypting data yet. The big thing with the cookie is just to be able to fetch data about a user's profile on the server. If someone wants to fake the cookie for some reason to load someone else's dashboard I think that's fine. They won't actually be able to do anything since access control is handled at the contract level anyway. Also they could just write their own UI to fetch anybody's dashboard.

**Comment by jonathanprozzi** [comment]
> > @baiirun
> > Regarding this TODO: "Move the data fetching for the user's profile to the server -- test out passing in connected user's address via cookie"
> > 
> > * I've been looking into/testing a few approaches that are cookie based, and have found a few examples of folks using Sign in with Ethereum (SIWE)+ Iron Session + app router to have the user data accessible on the server via cookies in a secure way
> > * Thinking if we don't want to use SIWE we could potentially use Iron Session to encrypt/decrypt the cookie with a secret
> > * Still doing a bit of testing with this since it's something we'll want to use in all user-scoped data fetching in the app router
> 
> Since all data in Geo is public I don't know if we need to worry too much about encrypting cookies yet. The big thing with the cookie is just to be able to fetch data about a user's profile on the server. If someone wants to fake the cookie for some reason to load someone else's dashboard I think that's fine. They won't actually be able to do anything since access control is handled at the contract level anyway. Also they could just write their own UI to fetch anybody's dashboard since all the data is available for anybody to query.
> 
> WRT to SIWE we probably want to have a broader conversation at some point around wallet onboarding which might include SIWE or some other mechanism.

Sounds good!

One thing I'm running into with the cookies is that `cookies.set()` seems to be either a route or server action, so now exploring the best way to set it. Once set we'll be able to use the `.get()` to get the user's address.

**Comment by baiirun** [comment]
> > > @baiirun
> > > Regarding this TODO: "Move the data fetching for the user's profile to the server -- test out passing in connected user's address via cookie"
> > > 
> > > * I've been looking into/testing a few approaches that are cookie based, and have found a few examples of folks using Sign in with Ethereum (SIWE)+ Iron Session + app router to have the user data accessible on the server via cookies in a secure way
> > > * Thinking if we don't want to use SIWE we could potentially use Iron Session to encrypt/decrypt the cookie with a secret
> > > * Still doing a bit of testing with this since it's something we'll want to use in all user-scoped data fetching in the app router
> > 
> > 
> > Since all data in Geo is public I don't know if we need to worry too much about encrypting cookies yet. The big thing with the cookie is just to be able to fetch data about a user's profile on the server. If someone wants to fake the cookie for some reason to load someone else's dashboard I think that's fine. They won't actually be able to do anything since access control is handled at the contract level anyway. Also they could just write their own UI to fetch anybody's dashboard since all the data is available for anybody to query.
> > WRT to SIWE we probably want to have a broader conversation at some point around wallet onboarding which might include SIWE or some other mechanism.
> 
> Sounds good!
> 
> One thing I'm running into with the cookies is that `cookies.set()` seems to be either a route or server action, so now exploring the best way to set it. Once set we'll be able to use the `.get()` to get the user's address. Should be able to be done with a server action I think

I think an API route to set the active account on the server makes sense. Also one to remove it when they log out.

**Comment by jonathanprozzi** [comment]
> > > > @baiirun
> > > > Regarding this TODO: "Move the data fetching for the user's profile to the server -- test out passing in connected user's address via cookie"
> > > > 
> > > > * I've been looking into/testing a few approaches that are cookie based, and have found a few examples of folks using Sign in with Ethereum (SIWE)+ Iron Session + app router to have the user data accessible on the server via cookies in a secure way
> > > > * Thinking if we don't want to use SIWE we could potentially use Iron Session to encrypt/decrypt the cookie with a secret
> > > > * Still doing a bit of testing with this since it's something we'll want to use in all user-scoped data fetching in the app router
> > > 
> > > 
> > > Since all data in Geo is public I don't know if we need to worry too much about encrypting cookies yet. The big thing with the cookie is just to be able to fetch data about a user's profile on the server. If someone wants to fake the cookie for some reason to load someone else's dashboard I think that's fine. They won't actually be able to do anything since access control is handled at the contract level anyway. Also they could just write their own UI to fetch anybody's dashboard since all the data is available for anybody to query.
> > > WRT to SIWE we probably want to have a broader conversation at some point around wallet onboarding which might include SIWE or some other mechanism.
> > 
> > 
> > Sounds good!
> > One thing I'm running into with the cookies is that `cookies.set()` seems to be either a route or server action, so now exploring the best way to set it. Once set we'll be able to use the `.get()` to get the user's address. Should be able to be done with a server action I think
> 
> I think an API route/server action to set the active account on the server makes sense. Also one to remove it when they log out.

Just tested this out in a fairly rudimentary way and it's working -- setting a cookie on successful connection for the active account and clearing it when they disconnect. Can refine, but I think this approach works if we turn on server actions in the config 👍

**Comment by jonathanprozzi** [comment]
@baiirun should we close this and then re-open it once ready to continue building it out? could also potentially merge in the UI partials and remove the Page/Component so there's no route? 

Happy to close & re-open when we revisit instead of having this unfinished PR hanging out!

**Comment by baiirun** [comment]
I'd advocate for just closing for now instead of partially merging things in

**Comment by jonathanprozzi** [comment]
Closing and will revisit once we resume building this feature.

**Pull request closed without merging**